### PR TITLE
Add e2e test for sequencer failover

### PR DESCRIPTION
**Pull request by anikaraghu**
**Description**

This PR has gotten pretty out of date so copying over the e2e test for sequencer failover work: https://github.com/ethereum-optimism/optimism/pull/8945

**Tests**

This PR adds a test

**Additional context**

**Metadata**

- Fixes #[Link to Issue]

### CODE REVIEW

File: op-e2e/sequencer_failover_test.go [hunk 1]
```diff
@@ -136,3 +136,31 @@ func TestSequencerFailover_ConductorRPC(t *testing.T) {
...
+	// The leadership change should occur with no errors
+	newID := waitForLeadershipChange(t, leader, leaderId, conductors, sys)
+	require.NotEqual(t, leaderId, newID, "Expected leader to change")
+
+	// Confirm the new leader is different from the old leader
+	newLeaderId, _ := findLeader(t, conductors)
+	require.NotEqual(t, leaderId, newLeaderId, "Expected leader to change")
+
+	// Check that the sequencer is healthy
+	require.True(t, healthy(t, ctx, conductors[newLeaderId]))
```
Comment (ajsutton) [inline op-e2e/sequencer_failover_test.go:160]: Is the sequencer guaranteed to be immediately healthy or do we need a waiter/eventually here?

**Review by ajsutton — COMMENTED** [review]
Look good, just want to confirm if the new leader is guaranteed to be immediately healthy.

File: op-e2e/sequencer_failover_test.go (hunk 1 above)
Comment (anikaraghu) [reply op-e2e/sequencer_failover_test.go:160]: @ajsutton yes it is guaranteed to be healthy if selected as the leader! https://github.com/ethereum-optimism/optimism/blob/develop/op-conductor/conductor/service.go#L550-L596

Comment (anikaraghu) [reply op-e2e/sequencer_failover_test.go:160]: yes @ajsutton in this case all but one were healthy and we waited for the leadership change from the 1 unhealthy one to one of the others, so the new leader here should always be healthy. 

Can also see here: https://github.com/ethereum-optimism/optimism/blob/develop/op-conductor/conductor/service.go#L550-L596

**Review by ajsutton — APPROVED** [review]

**Pull request merged**